camel/models/model_factory.py

```diff
@@ -155,3 +158,62 @@ def create(
             url=url,
             token_counter=token_counter,
         )
+
+    @classmethod
+    def __parse_model_platform(
+        cls, model_platform_str: str
+    ) -> ModelPlatformType:
+        try:
+            if model_platform_str.startswith("ModelPlatformType."):
+                platform_name = model_platform_str.split('.')[-1]
+            else:
+                platform_name = model_platform_str.upper()
+
+            if platform_name not in ModelPlatformType.__members__:
+                raise ValueError(
+                    f"Invalid model platform: {platform_name}. "
+                    f"Valid options: {', '.join(ModelPlatformType.__members__.keys())}"
```

Comment:
Hey @SaranshPandya,  

The pre-commit failure is throwing an **E501** error, which means the line is too long. You can fix this by applying the following suggestion:  

```suggestion
                    f"""Valid options: {', '
                    .join(ModelPlatformType.__members__.keys())}"""
```  

For next time, you can run:  

```
pre-commit run --all-files
```  

locally before committing and pushing your changes. This will help you catch and fix these issues before your commits. 🚀